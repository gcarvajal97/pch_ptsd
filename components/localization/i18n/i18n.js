export const fallback = "en";
 
export const supportedLocales = {
    en: {
        name: "English",
        translationFileLoader: () => require('../lang/en.json'),
        // en is default locale in Moment
        momentLocaleLoader: () => Promise.resolve(),
    },
    es: {
        name: "EspaÃ±ol",
        translationFileLoader: () => require('../lang/es.json'),
        momentLocaleLoader: () => import('moment/locale/es'),
    },
};
 
export const defaultNamespace = "LearnMore";
 
export const namespaces = [
    "LearnMore"
];