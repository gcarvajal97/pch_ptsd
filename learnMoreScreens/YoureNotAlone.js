import i18n from 'i18n-js';
import React, { Component } from "react";
import { View, StyleSheet, Text, ScrollView } from "react-native";
import { Video } from 'expo-av';
import translate, { getVideo } from '../components/translateService';
import colors from '../colors';

const videos = {
    en: require("../assets/videos/vidAlone.mp4"),
    es: require("../assets/videos/vidAloneEs.mp4"),
};

class YoureNotAlone extends Component {

    // Gives the NavBar a title for this specific screen
    static navigationOptions = {
        headerTitle: translate('youAreNotAlone.header')
    }
    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.title}>
                    {translate('youAreNotAlone.title')}
                </Text>
                <Video
                    source={getVideo(videos)}
                    rate={1.0}
                    volume={1.0}
                    resizeMode={Video.RESIZE_MODE_COVER}
                    shouldPlay
                    useNativeControls
                    style={{ 
                        height: 204, 
                        width: '100%', 
                        maxHeight: 254, 
                        borderColor: colors.accent, 
                        borderWidth: 14, 
                        borderTopWidth: 0, 
                        borderBottomWidth: 0, 
                        marginBottom:-1
                    }}
                    accessible
                    accessibilityLabel={translate(
                        'youAreNotAlone.content.videoCard.accessibility'
                        )}
                    accessibilityHint={translate(
                        'youAreNotAlone.content.videoCard.accessibility'
                        )}
                />
                <Text style={styles.subTitle}>
                    <Text style={{ fontFamily: 'avenir-heavy' }}>{translate('youAreNotAlone.content.videoCard.boldText')}</Text> 
                    {translate('youAreNotAlone.content.videoCard.subtitle')}
                </Text>
                <ScrollView>
                <Text style={{
                    fontFamily: 'avenir-heavy', 
                    fontSize:20, 
                    alignSelf:'stretch', 
                    textAlign:'left',
                    marginLeft:5, 
                    marginTop:4, 
                    marginBottom:3, 
                    color: colors.accent
                }}>
                    {translate('youAreNotAlone.content.statsBullets.intro')}
                    </Text>
                <Text style={styles.bullet}>{translate('youAreNotAlone.content.statsBullets.bulletOne')}</Text>
                <Text style={styles.bullet}>{translate('youAreNotAlone.content.statsBullets.bulletTwo')}</Text>
                <Text style={styles.bullet}>{translate('youAreNotAlone.content.statsBullets.bulletThree')}</Text>
                <Text style={styles.paragraph}>{translate('youAreNotAlone.content.paragraphOne')}</Text>
                <Text style={styles.paragraph}>{translate('youAreNotAlone.content.paragraphTwo')}</Text>        
                </ScrollView>
            </View>
        );
    }
}
export default YoureNotAlone;

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      margin: 10,
      backgroundColor: colors.primary.white,
      shadowColor: colors.primary.black,
      shadowOffset: {
          width: 0,
          height: 2,
      },
      shadowOpacity: 0.23,
      shadowRadius: 2.62,

      elevation: 4,
  },
  title: {
      backgroundColor: colors.accent, 
      color: colors.primary.white, 
      fontFamily: 'avenir-heavy', 
      fontSize: 22, 
      textAlign: 'center', 
      alignSelf: 'stretch'
  },
  subTitle: {
      backgroundColor: colors.accent, 
      color: colors.primary.white, 
      alignSelf: 'stretch', 
      paddingLeft: 15, 
      paddingRight: 15, 
      fontSize: 15, 
      borderBottomWidth: 2, 
      borderBottomColor: colors.accent,
  },
  paragraphTitle: {
      fontFamily: 'avenir-heavy',
      fontSize: 20,
      alignSelf: 'stretch',
      textAlign: 'left',
      marginLeft: 5,
      marginTop: 4,
      marginBottom: 3,
      color: colors.accent
  },
  bullet: {
      marginVertical: 1, 
      marginHorizontal: 18, 
      textAlign: 'left', 
      alignSelf: 'stretch', 
      fontFamily: 'avenir-heavy',
      fontSize: 14
  },
  paragraph: {
      alignSelf: "center", 
      paddingVertical: 1, 
      paddingHorizontal: 5,
      fontSize: 14,
      marginTop: 5,
  },
  link: {
      alignSelf:"center", 
      paddingVertical: 1, 
      paddingHorizontal: 5,
      fontSize: 16,
      color: colors.accent,
      textDecorationLine: 'underline',
      marginTop: 4,
      marginTop: 3,
  }
});