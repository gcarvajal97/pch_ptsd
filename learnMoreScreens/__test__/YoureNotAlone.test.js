import i18n from 'i18n-js';
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from 'react';
import YoureNotAlone from '../YoureNotAlone';
import { Video } from 'expo-av';

import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe("YoureNotAlone", () => {
    let wrapper;
    beforeEach(() => {
        i18n.locale = "en";
    });

    it('renders correctly', () => {
            const tree = renderer.create(<YoureNotAlone />).toJSON();
            expect(tree).toMatchSnapshot();}
        )

     it('renders correctly in Spanish', () => {
            i18n.locale = 'es'
            const tree = renderer.create(<YoureNotAlone />).toJSON();
            expect(tree).toMatchSnapshot();}
        )
})