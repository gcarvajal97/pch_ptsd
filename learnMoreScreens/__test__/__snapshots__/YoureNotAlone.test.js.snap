// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoureNotAlone renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "elevation": 4,
      "flex": 1,
      "margin": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#0077C8",
        "color": "#FFFFFF",
        "fontFamily": "avenir-heavy",
        "fontSize": 22,
        "textAlign": "center",
      }
    }
  >
    You Are Not Alone
  </Text>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#0077C8",
        "borderTopWidth": 0,
        "borderWidth": 14,
        "height": 204,
        "marginBottom": -1,
        "maxHeight": 254,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoVideoView
      accessibilityHint="Video on how injury is common"
      accessibilityLabel="Video on how injury is common"
      accessible={true}
      proxiedProperties={
        Object {
          "onError": [Function],
          "onFullscreenUpdate": [Function],
          "onLoad": [Function],
          "onLoadStart": [Function],
          "onReadyForDisplay": [Function],
          "onStatusUpdate": [Function],
          "resizeMode": "mock",
          "source": Object {
            "headers": undefined,
            "overridingExtension": null,
            "uri": "uri",
          },
          "status": Object {
            "rate": 1,
            "shouldPlay": true,
            "volume": 1,
          },
          "useNativeControls": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#0077C8",
        "borderBottomColor": "#0077C8",
        "borderBottomWidth": 2,
        "color": "#FFFFFF",
        "fontSize": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "avenir-heavy",
        }
      }
    >
      Your child is not alone.
    </Text>
     Every year, millions of children in the United States are injured.
  </Text>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#0077C8",
            "fontFamily": "avenir-heavy",
            "fontSize": 20,
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        Each year in the United States, child injury statistics show that:
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "avenir-heavy",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - 1 in 4 children and teens need medical care for an injury.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "avenir-heavy",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - 7 million injured children are treated in the ER.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "avenir-heavy",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Several hundred thousand injured children are admitted to the hospital.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        Right now, hundreds of thousands of children across the country are recovering from an injury. Like you, their parents and other caring adults want to help them the best way possible. While it is important to tend to the wounds and rehabilitation, it is just as important to remember to look beyond the physical injuries.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        With all the doctors, nurses, and therapists who will treat your child’s injury, remember that no one is more important than you for your child’s recovery. In making this application, we took the best information available from science and best practices to help you and your child recover after an injury.
      </Text>
    </View>
  </RCTScrollView>
</View>
`;

exports[`YoureNotAlone renders correctly in Spanish 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "elevation": 4,
      "flex": 1,
      "margin": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#0077C8",
        "color": "#FFFFFF",
        "fontFamily": "avenir-heavy",
        "fontSize": 22,
        "textAlign": "center",
      }
    }
  >
    No Está Solo
  </Text>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#0077C8",
        "borderTopWidth": 0,
        "borderWidth": 14,
        "height": 204,
        "marginBottom": -1,
        "maxHeight": 254,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoVideoView
      accessibilityHint="Video sobre cómo las lesiones son comunes"
      accessibilityLabel="Video sobre cómo las lesiones son comunes"
      accessible={true}
      proxiedProperties={
        Object {
          "onError": [Function],
          "onFullscreenUpdate": [Function],
          "onLoad": [Function],
          "onLoadStart": [Function],
          "onReadyForDisplay": [Function],
          "onStatusUpdate": [Function],
          "resizeMode": "mock",
          "source": Object {
            "headers": undefined,
            "overridingExtension": null,
            "uri": "uri",
          },
          "status": Object {
            "rate": 1,
            "shouldPlay": true,
            "volume": 1,
          },
          "useNativeControls": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#0077C8",
        "borderBottomColor": "#0077C8",
        "borderBottomWidth": 2,
        "color": "#FFFFFF",
        "fontSize": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "avenir-heavy",
        }
      }
    >
      Su hijo no está solo.
    </Text>
     Cada año, milliones de niños en los Estados Unidos resultan heridos.
  </Text>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#0077C8",
            "fontFamily": "avenir-heavy",
            "fontSize": 20,
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        Cada año en EE. UU., las estadísticas de lesiones infantiles muestran que:
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "avenir-heavy",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - 1 de cada 4 niños y adolescentes requiere atención médica por una lesión
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "avenir-heavy",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - 7 millones de niños heridos son tratados en salas de emergencias.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontFamily": "avenir-heavy",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Cientos de miles de niños heridos son hospitalizados.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        En este momento, cientos de miles de niños de todo el país se están recuperando de una lesión. Al igual que usted, sus padres y otros adultos quieren ayudarlos de la mejor manera posible. Si bien es importante atender las heridas y ocuparse de la rehabilitación, es igual de importante recordar que hay que mirar más allá de las lesiones físicas.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        Aunque haya muchos médicos, enfermeras y terapeutas que traten la lesión de su hijo, recuerde que nadie es más importante que usted para la recuperación de su hijo. En este sitio web tomamos lo mejor de la ciencia y la práctica para ayudarle a ayudar a su hijo después de una lesión.
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
